### PR TITLE
Add MITRE groups endpoint

### LINKED ISSUE #7819: Create new MITRE endpoints
Hello team,

Due to the changes in MITRE db performed in https://github.com/wazuh/wazuh/issues/7607, we should adapt the framework to the changes and create new API endpoints to access all the MITRE information.

As a result of this epic, we should have at least endpoints to access the following information:

- MITRE metadata.
- Tactics and their related techniques.
- Techniques and their related tactics, groups, software and mitigations.
- Mitigations and their related techniques.
- Groups and their related techniques.
- Software and their related techniques.
- References.

Regards,

David J. Iglesias

**WORKING_BRANCH**: https://github.com/wazuh/wazuh/tree/feature/7819-new-endpoints-mitre
**TARGET_BRANCH**: https://github.com/wazuh/wazuh/tree/dev-mitredb
**DOCUMENTATION_BRANCH**: https://github.com/wazuh/wazuh-documentation/tree/feature/7819-new-endpoints-mitre

Latest rebase with **TARGET_BRANCH**: April, 13th - 17:00 CET

### LINKED ISSUE #7827: Develop new MITRE group endpoints
|Wazuh version|Component|
|---|---|
| N/A | API/Framework |

## Description

Hello team,

This issue focuses on the development and implementation of the new MITRE `group` framework, API endpoints and all the necessary tests.

Regards,

Adrián Peña

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

### PR DESCRIPTION
**Pull request by vicferpoy**
|Related issue|
|---|
| [7827](https://github.com/wazuh/wazuh/issues/7827) |

Hello team,
this PR closes #7827   .

This PR adds a new endpoint for MITRE groups and its unit and integration tests. In addition, I have fixed some errors in the tests that were including in previous developments and typos.

### Unit tests

```
==================================================================================== test session starts ====================================================================================
platform linux -- Python 3.9.1, pytest-5.4.3, py-1.8.2, pluggy-0.13.1
rootdir: /home/vicferpoy/Desktop/Git/wazuh/framework
plugins: asyncio-0.14.0, metadata-1.10.0
collected 1549 items                                                                                                                                                                        

wazuh/core/cluster/dapi/tests/test_dapi.py .......................                                                                                                                    [  1%]
wazuh/core/cluster/tests/test_cluster.py ...................                                                                                                                          [  2%]
wazuh/core/cluster/tests/test_common.py .......                                                                                                                                       [  3%]
wazuh/core/cluster/tests/test_control.py .....                                                                                                                                        [  3%]
wazuh/core/cluster/tests/test_local_client.py .                                                                                                                                       [  3%]
wazuh/core/cluster/tests/test_utils.py .......                                                                                                                                        [  4%]
wazuh/core/cluster/tests/test_worker.py ................                                                                                                                              [  5%]
wazuh/core/tests/test_active_response.py ..............                                                                                                                               [  5%]
wazuh/core/tests/test_agent.py ...................................................................................................................................................... [ 15%]
.......                                                                                                                                                                               [ 16%]
wazuh/core/tests/test_cdb_list.py ......................................                                                                                                              [ 18%]
wazuh/core/tests/test_common.py .......                                                                                                                                               [ 18%]
wazuh/core/tests/test_configuration.py ....................................                                                                                                           [ 21%]
wazuh/core/tests/test_decoder.py ................                                                                                                                                     [ 22%]
wazuh/core/tests/test_input_validator.py ...                                                                                                                                          [ 22%]
wazuh/core/tests/test_logtest.py ..                                                                                                                                                   [ 22%]
wazuh/core/tests/test_manager.py .................                                                                                                                                    [ 23%]
wazuh/core/tests/test_mitre.py .......                                                                                                                                                [ 24%]
wazuh/core/tests/test_pyDaemonModule.py ......                                                                                                                                        [ 24%]
wazuh/core/tests/test_results.py ........................................                                                                                                             [ 27%]
wazuh/core/tests/test_rootcheck.py ..........                                                                                                                                         [ 27%]
wazuh/core/tests/test_rule.py ......................                                                                                                                                  [ 29%]
wazuh/core/tests/test_stats.py ....                                                                                                                                                   [ 29%]
wazuh/core/tests/test_syscollector.py ...                                                                                                                                             [ 29%]
wazuh/core/tests/test_utils.py .................XXx.................................................................................................................................. [ 39%]
.....................................................................                                                                                                                 [ 43%]
wazuh/core/tests/test_vulnerability.py .                                                                                                                                              [ 43%]
wazuh/core/tests/test_wazuh_queue.py ..................                                                                                                                               [ 45%]
wazuh/core/tests/test_wazuh_socket.py ....................                                                                                                                            [ 46%]
wazuh/core/tests/test_wdb.py .....................                                                                                                                                    [ 47%]
wazuh/rbac/tests/test_auth_context.py ..                                                                                                                                              [ 47%]
wazuh/rbac/tests/test_decorators.py .........................................................................................................                                         [ 54%]
wazuh/rbac/tests/test_default_configuration.py .......................................................                                                                                [ 58%]
wazuh/rbac/tests/test_orm.py ......................................................                                                                                                   [ 61%]
wazuh/rbac/tests/test_preprocessor.py ...........                                                                                                                                     [ 62%]
wazuh/tests/test_active_response.py ............                                                                                                                                      [ 63%]
wazuh/tests/test_agent.py ...................................................................................................                                                         [ 69%]
wazuh/tests/test_cdb_list.py .....................................................                                                                                                    [ 72%]
wazuh/tests/test_ciscat.py .................................                                                                                                                          [ 75%]
wazuh/tests/test_cluster.py .......                                                                                                                                                   [ 75%]
wazuh/tests/test_decoder.py ...................................                                                                                                                       [ 77%]
wazuh/tests/test_group.py ........                                                                                                                                                    [ 78%]
wazuh/tests/test_logtest.py ......                                                                                                                                                    [ 78%]
wazuh/tests/test_manager.py .................................                                                                                                                         [ 80%]
wazuh/tests/test_mitre.py ....                                                                                                                                                        [ 81%]
wazuh/tests/test_rootcheck.py .................................................                                                                                                       [ 84%]
wazuh/tests/test_rule.py ........................................................                                                                                                     [ 87%]
wazuh/tests/test_sca.py .......                                                                                                                                                       [ 88%]
wazuh/tests/test_security.py ...........................................................................                                                                              [ 93%]
wazuh/tests/test_stats.py ................                                                                                                                                            [ 94%]
wazuh/tests/test_syscheck.py ........................                                                                                                                                 [ 95%]
wazuh/tests/test_syscollector.py ............                                                                                                                                         [ 96%]
wazuh/tests/test_task.py ............................                                                                                                                                 [ 98%]
wazuh/tests/test_vulnerability.py ..........................                                                                                                                          [100%]

============================================================ 1546 passed, 1 xfailed, 2 xpassed, 14 warnings in 331.03s (0:05:31) ============================================================

```

### Integration tests

```
Collected tests [4]:
test_mitre_endpoints.tavern.yaml, test_rbac_black_mitre_endpoints.tavern.yaml, test_rbac_white_all_endpoints.tavern.yaml, test_rbac_white_mitre_endpoints.tavern.yaml


test_mitre_endpoints.tavern.yaml 
	 4 passed, 8 warnings

test_rbac_black_mitre_endpoints.tavern.yaml 
	 4 passed, 8 warnings

test_rbac_white_all_endpoints.tavern.yaml 
	 153 passed, 3 xfailed, 158 warnings

test_rbac_white_mitre_endpoints.tavern.yaml 
	 4 passed, 8 warnings

```

Regards,
Víctor Fernández

### CODE REVIEW

**Review by davidjiglesias — CHANGES_REQUESTED** [review]

File: api/api/spec/spec.yaml [hunk 1]
```diff
@@ -10719,6 +10730,91 @@ paths:
       responses:
         '200':
           description: "Get MITRE techniques information"
+          content:
```
[the remark this answers is not in the document]
Comment (vicferpoy) [reply api/api/spec/spec.yaml:10733]: This got messed up in the rebase. Fixed.

File: api/api/spec/spec.yaml [hunk 2]
```diff
@@ -1724,6 +1724,9 @@ components:
     Tactic_id:
       type: string
       description: "Tactic ID"
+    Mitre_group_id:
```
[the remark this answers is not in the document]
Comment (vicferpoy) [reply api/api/spec/spec.yaml:1727]: We discussed this. Not applying it in this PR.

File: api/api/spec/spec.yaml [hunk 3]
```diff
@@ -10719,6 +10730,91 @@ paths:
...
+        - $ref: '#/components/parameters/wait_for_complete'
+        - $ref: '#/components/parameters/offset'
+        - $ref: '#/components/parameters/limit'
+        - $ref: '#/components/parameters/sort'
+        - $ref: '#/components/parameters/search'
+        - $ref: '#/components/parameters/select'
+        - $ref: '#/components/parameters/query'
+      responses:
+        '200':
+          description: "Get MITRE groups information"
```
[the remark this answers is not in the document]
Comment (vicferpoy) [reply api/api/spec/spec.yaml:10817]: We discussed this. Not applying it in this PR.

File: api/test/integration/test_mitre_endpoints.tavern.yaml [hunk 4]
```diff
@@ -537,25 +537,13 @@ stages:
...
-              related_groups: !anything
-            - <<: *techniques_response
           failed_items: []
           total_affected_items: !anyint
           total_failed_items: 0
+      verify_response_with:
+        function: tavern_utils:test_validate_search
+        extra_kwargs:
+          search_param: 'Gather Victim Network Information'
       save:
```
[the remark this answers is not in the document]
Comment (vicferpoy) [reply api/test/integration/test_mitre_endpoints.tavern.yaml:547]: Done.

File: api/test/integration/test_rbac_black_mitre_endpoints.tavern.yaml [hunk 5]
```diff
@@ -56,3 +56,19 @@ stages:
...
     response:
       <<: *permission_denied
+
+---
+test_name: GET /mitre/groups
+
+stages:
+
+  # GET /mitre/groups
+  - name: Try to request MITRE groups (Denied)
```
[the remark this answers is not in the document]
Comment (vicferpoy) [reply api/test/integration/test_rbac_black_mitre_endpoints.tavern.yaml:66]: Done.

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**